src/chainstate/stacks/boot/bns.clar

```diff
@@ -0,0 +1,973 @@
+  
+(define-private (get-exp-at-index (buckets (list 16 uint)) (index uint))
+  (get value (fold element-at buckets (tuple (limit index) (cursor u0) (value u0)))))
+
+(define-private (is-digit (char (buff 1)))
+  (or 
+    (is-eq char 0x30) ;; 0
+    (is-eq char 0x31) ;; 1
+    (is-eq char 0x32) ;; 2
+    (is-eq char 0x33) ;; 3
+    (is-eq char 0x34) ;; 4
+    (is-eq char 0x35) ;; 5
+    (is-eq char 0x36) ;; 6
+    (is-eq char 0x37) ;; 7
+    (is-eq char 0x38) ;; 8
+    (is-eq char 0x39))) ;; 9
+
+(define-private (is-lowercase-alpha (char (buff 1)))
+  (or 
+    (is-eq char 0x61) ;; a
+    (is-eq char 0x62) ;; b
+    (is-eq char 0x63) ;; c
+    (is-eq char 0x64) ;; d
+    (is-eq char 0x65) ;; e
+    (is-eq char 0x66) ;; f
+    (is-eq char 0x67) ;; g
+    (is-eq char 0x68) ;; h
+    (is-eq char 0x69) ;; i
+    (is-eq char 0x6a) ;; j
+    (is-eq char 0x6b) ;; k
+    (is-eq char 0x6c) ;; l
+    (is-eq char 0x6d) ;; m
+    (is-eq char 0x6e) ;; n
+    (is-eq char 0x6f) ;; o
+    (is-eq char 0x70) ;; p
+    (is-eq char 0x71) ;; q
+    (is-eq char 0x72) ;; r
+    (is-eq char 0x73) ;; s
+    (is-eq char 0x74) ;; t
+    (is-eq char 0x75) ;; u
+    (is-eq char 0x76) ;; v
+    (is-eq char 0x77) ;; w
+    (is-eq char 0x78) ;; x
+    (is-eq char 0x79) ;; y
+    (is-eq char 0x7a))) ;; z
+
+(define-private (is-vowel (char (buff 1)))
+  (or 
+    (is-eq char 0x61) ;; a
+    (is-eq char 0x65) ;; e
+    (is-eq char 0x69) ;; i
+    (is-eq char 0x6f) ;; o
+    (is-eq char 0x75) ;; u
+    (is-eq char 0x79))) ;; y
+
+(define-private (is-special-char (char (buff 1)))
+  (or 
+    (is-eq char 0x2d) ;; -
+    (is-eq char 0x5f))) ;; _
+
+(define-private (is-char-valid (char (buff 1)))
+  (or 
+    (is-lowercase-alpha char)
+    (is-digit char)
+    (is-special-char char)))
+
+(define-private (is-nonalpha (char (buff 1)))
+  (or 
+    (is-digit char)
+    (is-special-char char)))
+
+(define-private (has-vowels-chars (name (buff 16)))
+  (> (len (filter is-vowel name)) u0))
+
+(define-private (has-nonalpha-chars (name (buff 16)))
+  (> (len (filter is-nonalpha name)) u0))
+
+(define-private (has-invalid-chars (name (buff 19)))
+  (< (len (filter is-char-valid name)) (len name)))
+
+(define-private (compute-name-price (name (buff 16))
+                                    (price-function (tuple (buckets (list 16 uint)) (base uint) (coeff uint) (nonalpha-discount uint) (no-vowel-discount uint))))
+  (let (
+    (exponent (get-exp-at-index (get buckets price-function) (min u15 (- (len name) u1))))
+    (no-vowel-discount (if (not (has-vowels-chars name)) (get no-vowel-discount price-function) u1))
+    (nonalpha-discount (if (has-nonalpha-chars name) (get nonalpha-discount price-function) u1)))
+    (*
+      (/
+        (*
+          (get coeff price-function)
+          (pow (get base price-function) exponent))
+        (max nonalpha-discount no-vowel-discount))
+      u10))) ;; 10 = name_cost (100) * "old_price_multiplier" (0.1) - todo(ludo): sort this out.
```

Comment:
Can you elaborate more on this `todo`?

Reply:
Yeah I think my question is around the value `u10`.
If I remember well, it comes from a formula that I've found in the python code, does we need to refresh that?

Reply:
No, I think that's fine.  I would just add that as a comment -- this is a legacy hold-over from the Python implementation.